plugin/src/main/java/com/playmonumenta/scriptedquests/listeners/PlayerListener.java

```diff
@@ -180,4 +234,35 @@ public void playerQuitEvent(PlayerQuitEvent event) {
 		// Remove all zone properties from the player
 		mPlugin.mZoneManager.unregisterPlayer(event.getPlayer());
 	}
+
+	private class EnableLeftClick extends BukkitRunnable {
+
+		private Player mPlayer;
+
+		@Override
+		public void run() {
+			mPlayer.removeScoreboardTag("usedinteractable");
+		}
+
+	}
+
+	private class DelayLeftClick extends BukkitRunnable {
+
+		private PlayerAnimationEvent mEvent;
+
+		@Override
+		public void run() {
+
+			//If the player recently used a right click or left click air
+			if (mEvent.getPlayer().getScoreboardTags().contains("usedinteractable")) {
+				mEvent.setCancelled(true);
+			} else {
+				//If not, must be a left click block. Run that function
+				delayedAnimationEvent(mEvent);
+			}
+		}
+
+	}
```

Comment:
This runnable isn't needed, and also you can't cancel an event later. You have to cancel it before the event callback returns.